pkg/ccl/backupccl/restore_planning.go

```diff
@@ -1486,13 +1486,13 @@ func doRestorePlan(
 		defaultURIs, mainBackupManifests, localityInfo, memReserved, err = resolveBackupManifestsExplicitIncrementals(
 			ctx, &mem, mkStore, from, endTime, encryption, p.User())
 	}
+	defer func() {
```

Comment:
I think this may have been better where it was? I think it's more conventional to only dall the `defer` if the allocation of resources was successful, i.e. err==nil. But maybe I'm mistaken!

Reply:
i moved it back, partially to keep this backport to show.go stuff.

I don't think it really matters and I don't know enough about our memory monitoring system to have a confident opinion :D. But, I initially changed the defer func location because doRestorePlan assumes that resolveBackupManifestsExplicitIncrementals will only return a nonzero memreserved if err == nil, which is currently the case. But that may not always be the case in the future.

Reply:
Yes, that's true. But the worst outcome would be to run a cleanup on a resource that doesn't exist, potentially dereferencing a nil pointer, so I lean towards the "functions that error clean up after themselves" paradigm.

It for sure doesn't matter because in this particular case memReserved will just be zero if err != nil. :) Perusing the Internet, I'm surprised to find that both ways are actually used. That doesn't make much sense to me, but I'm hardly a Go expert 🤷  TIL!